### PR TITLE
Use NewRegistration instead of TransientRegistration when appropriate

### PR DESCRIPTION
**Pull request by irisfaraway**
https://eaflood.atlassian.net/browse/RUBY-62

The engine now allows for multiple types of TransientRegistrations, starting with a NewRegistration.

The back office has generally been assuming that TransientRegistrations are only for the new registration journey. Now that we have the NewRegistration class specifically for that, we need to update the back office to use it.

### CODE REVIEW

**Comment by irisfaraway** [comment]
The Gemfile will need to be changed back to `master` once https://github.com/DEFRA/waste-exemptions-engine/pull/138 is merged in.

**Comment by irisfaraway** [comment]
@Cruikshanks @cintamani I've requested a review for this as well, but it will be voided once https://github.com/DEFRA/waste-exemptions-engine/pull/138 is merged so you may need to re-review again. If you'd rather wait, that is cool 👍

File: Gemfile [hunk 1]
```diff
@@ -62,7 +62,7 @@ gem "paper_trail", "~> 10.2.0"
 # Use the waste exemptions engine for the user journey
 gem "waste_exemptions_engine",
     git: "https://github.com/DEFRA/waste-exemptions-engine",
-    branch: "master"
+    branch: "feature/transient-reg-subclasses"
```
Comment (cintamani) [inline Gemfile]: I don't think this should be part of the changes right? :)

Comment (irisfaraway) [reply Gemfile]: Nope, I'll have to rip it out once the engine update is merged!

**Comment by irisfaraway** [comment]
@cintamani @Cruikshanks Now that the first batch of engine changes have been merged in, I've switched the branch back to master here – unfortunately that has voided all the reviews. Would you mind having another quick look?

**Pull request merged**